### PR TITLE
added recursive method for palindrome

### PR DESCRIPTION
**Pull request by nischal20**

### CODE REVIEW

**Review by siriak — APPROVED** [review]

File: strings/palindrome/ispalindrome.go [hunk 1]
```diff
@@ -35,3 +35,23 @@ func IsPalindrome(text string) bool {
...
+			return false
+		}
+	} else {
+		return true
+	}
+	start = start + 1
+	end = end - 1
+
+	return IsPalindromeRecursiveHelper(runes, start, end)
+}
```
Comment (tjgurwara99) [inline strings/palindrome/ispalindrome.go:57]:
```suggestion
func IsPalindromeRecursiveHelper(runes []rune, start int64, end int64) bool {
	if start >= end {
		return true
	}
	if string(runes[start]) != string(runes[end-1]) {
		return false
	}
	start = start + 1
	end = end - 1

	return IsPalindromeRecursiveHelper(runes, start, end)
}
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: strings/palindrome/ispalindrome.go (hunk 1 above)
Comment (tjgurwara99) [reply strings/palindrome/ispalindrome.go:57]: If you know the condition to return then return early.

File: strings/palindrome/ispalindrome.go [hunk 2]
```diff
@@ -35,3 +35,22 @@ func IsPalindrome(text string) bool {
...
+	clean_text := cleanString(text)
+	runes := []rune(clean_text)
+	return IsPalindromeRecursiveHelper(runes, 0, int64(len(runes)))
+}
+
+func IsPalindromeRecursiveHelper(runes []rune, start int64, end int64) bool {
+	if start >= end {
+		return true
+	}
+	if string(runes[start]) != string(runes[end-1]) {
```
Comment (tjgurwara99) [inline strings/palindrome/ispalindrome.go:49]: The string conversion is redundant here

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: strings/palindrome/ispalindrome.go (hunk 2 above)
Comment (nischal20) [reply strings/palindrome/ispalindrome.go]: removed string conversion

File: strings/palindrome/ispalindrome.go [hunk 3]
```diff
@@ -35,3 +35,21 @@ func IsPalindrome(text string) bool {
...
 	return true
 }
+
+func IsPalindromeRecursive(text string) bool {
+	clean_text := cleanString(text)
+	runes := []rune(clean_text)
+	return IsPalindromeRecursiveHelper(runes, 0, int64(len(runes)))
+}
+
+func IsPalindromeRecursiveHelper(runes []rune, start int64, end int64) bool {
```
Comment (tjgurwara99) [inline strings/palindrome/ispalindrome.go:45]:
```suggestion
func isPalindromeRecursiveHelper(runes []rune, start int64, end int64) bool {
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]
Make the helper an unexported function. That is not useful/needed outside this package.

**Review by tjgurwara99 — DISMISSED** [review]
LGTM

**Review by siriak — APPROVED** [review]

**Review by tjgurwara99 — APPROVED** [review]

**Pull request merged**